addon/components/base/bs-form.js

```diff
@@ -242,6 +243,24 @@ export default Component.extend({
     }
   },
 
+  _validate() {
+    let model = this.get('model');
+
+    let validationPromise = this.validate(model);
+    if (validationPromise && validationPromise instanceof RSVP.Promise) {
+      let validatedPromise = new RSVP.Promise((resolve, reject) => {
+        validationPromise
+          .then(() => { resolve() })
+          .catch(() => {
+            this.set('showAllValidations', true);
+            return reject();
```

Comment:
I didn't add the `onInvalid` call here that exists in `submit`, which might not match what we're after. It seemed like the caller might want more control than that, but happy to put that in if you'd like